components/public-api/gitpod/v1/workspaces.proto

```diff
@@ -0,0 +1,322 @@
+}
+message ListWorkspacesResponse {
+    message WorkspaceAndInstance {
+        Workspace result = 1;
+        WorkspaceInstance last_active_instances = 2;
+    }
+
+    google.rpc.Status response_status = 1;
+
+    string next_page_token = 2;
+
+    repeated WorkspaceAndInstance result = 3;
+}
+
+message GetWorkspaceRequest {
+    string workspace_id = 1;
+}
+message GetWorkspaceResponse {
+    google.rpc.Status response_status = 1;
+
+    Workspace result = 2;
+}
+
+message CreateAndStartWorkspaceRequest {
+    string idempotency_token = 1;
+
+    oneof source {
+        string context_url = 2;
+        string prebuild_id = 3;
+    }
+
+    StartWorkspaceSpec start_spec = 5;
+}
+message CreateAndStartWorkspaceResponse {
+    google.rpc.Status response_status = 1;
+
+    string workspace_id = 2;
+}
+
+message StartWorkspaceRequest {
+    string idempotency_token = 1;
+
+    string workspace_id = 2;
+    StartWorkspaceSpec spec = 3;
+}
+message StartWorkspaceResponse {
+    google.rpc.Status response_status = 1;
+
+    string instance_id = 2;
+    string workspace_url = 3;
+}
+
+message GetActiveWorkspaceInstanceRequest {
+    string workspace_id = 1;
+}
+message GetActiveWorkspaceInstanceResponse {
+    google.rpc.Status response_status = 1;
+
+    WorkspaceInstance instance = 2;
+}
+
+message GetWorkspaceInstanceOwnerTokenRequest {
+    string instance_id = 1;
+}
+message GetWorkspaceInstanceOwnerTokenResponse {
+    google.rpc.Status response_status = 1;
+
+    string owner_token = 2;
+}
+
+message ListenToWorkspaceInstanceRequest {
+    string instance_id = 1;
+}
+message ListenToWorkspaceInstanceResponse {
+    google.rpc.Status response_status = 1;
+
+    WorkspaceInstanceStatus instance_status = 2;
+}
+
+message ListenToImageBuildLogsRequest {
+    string instance_id = 1;
+}
+message ListenToImageBuildLogsResponse {
+    google.rpc.Status response_status = 1;
+
+    string line = 2;
+}
+
+message StopWorkspaceRequest {
+    string idempotency_token = 1;
+
+    string workspace_id = 2;
+}
+message StopWorkspaceResponse {
+    google.rpc.Status response_status = 1;
+}
+
+////////////////////////////////
+// Shared messages come here
+////////////////////////////////
+
+// Workspace describes a single workspace
+message Workspace {
+    // workspace_id is the ID of the workspace
+    string workspace_id = 1;
+
+    // owner_id is the ID of the user who created this workspace
+    string owner_id = 2;
+
+    // project_id is the ID of the project which this workspace belongs to
+    string project_id = 3;
+
+    // context reports the original context the workspace was created from
+    WorkspaceContext context = 4;
+
+    // description is a human readable description of the workspace
+    string description = 5;
+}
+
+// WorkspaceContext describes the context a workspace was created from
+message WorkspaceContext {
+    // Explicit Git context
+    message Git {
+        string normalized_context_url = 1;
```

Comment:
what is the normalized context url? 
let's use the clone url instead.
I just found out, that some providers might use different url patterns in comparison to the context url, so you cannot just compute the clone url from it, instead it's some result from external api services which we need to store.